### ISSUE TITLE
How to configure, make & install on a Debian 10 system?

### ISSUE THREAD

**Issue by XtoreX** [rescuezilla/rescuezilla#247]
Good nights, my dear friends.
I have a PC with nine partitions on GPT (Debian, Mint, Ubuntu Studio, etc...) and the first partition has a Debian with de GRUB installed on it.
This is my especial BOOT Control SO, includes GRUB Customizer, GRUB Rescue, and now I want to install your RescueZilla but, don't know how to install from the ".tar.gz" file without some dependencies, because the "_all.deb" file generate errors about dependencies not installed.

Could you help me about how can do it?

**Comment by shasheene** [comment]
Hi @XtoreX,

There are two approaches.

The first is to [install the Rescuezilla ISO image as a rescue partition](https://github.com/rescuezilla/rescuezilla/wiki/Installing-Rescuezilla-as-a-rescue-partition).

The second approach is to install the [standalone Rescuezilla deb file using `gdebi`](https://github.com/rescuezilla/rescuezilla/wiki/Installing-Rescuezilla-from-the-standalone-deb-package). The `gdebi` tool is used to easily resolve dependencies.

There are certain circumstances where installing GRUB as a rescue partition or standalone deb file makes sense, but **it's definitely not the best approach in most circumstances. Rescuezilla is designed and recommended to be used as a live USB stick**. For user safety there are still some restrictions preventing the source and destination partitions residing on the same disk. 

There are circumstances where using Rescuezilla to a restore a partition lead to the re-installing of the GRUB bootloader that's present on the destination hard drive. While this is a safe operation, it's important that users are easily able to boot back into Rescuezilla in the unlikely event something goes wrong. **So unless there's a specific reason, I highly recommend booting Rescuezilla as a live USB stick.**

**Comment by XtoreX** [comment]
Hello, my friend, thanks for your time, but my question is "how to configure, make & install on a Debian 10 system?".

Your first approach, are replace my Debian 10, with the Rescuezilla ISO fresh install, and that's not a solution. I have installed clonezilla in my Debian 10, my purpouse is install a GUI in XFCE, as educational personal project.

The second approach, finally success. Was made with "dpkg -i", with errors, then a necessary "apt --fix-broken install" and re-do "dpkg -i". And that's all.

Only have an issue, I need to run "sudo rescuezilla" in an terminal emulator, because the Panel/Application don't works.

I'll keep investigating about this.

I'll close the question issue, thanks.

**Comment by shasheene** [comment]
By the way the first approach does _not_ replace your Debian 10 but it installs the Rescuezilla ISO image along side your existing GRUB boot entries.

But yes, your approach using `apt --fix-broken install` is a common way of installing dependencies associated with a deb package. The alternative approach is to first install `gdebi` and then use that to install the deb file.

> Only have an issue, I need to run "sudo rescuezilla" in an terminal emulator, because the Panel/Application don't works.

By the way, you can create a symlink to create a Desktop entry using the following:

`sudo ln -s /usr/share/applications/rescuezilla.desktop "/home/XtoreX/Desktop/rescuezilla.desktop"`

To create a start menu item, there is a Free Desktop Menu Spec XML configuration file. But depending on the environment there is often a graphical frontend that you can use to modify the menu without need to manually modify the XML file. But the _most_ correct solution may be for me to modify the `/usr/share/applications/rescuezilla.desktop` to associate the correct tags etc so it automatically gets populated into the Panel/Application start menu given your system's existing Free Desktop Menu Spec configuration.

**Comment by candrapersada** [comment]
> Good nights, my dear friends.
> I have a PC with nine partitions on GPT (Debian, Mint, Ubuntu Studio, etc...) and the first partition has a Debian with de GRUB installed on it.

why by default the Overwrite partition table is selected? backup 2 linux partitions from 9 linux partitions
![A](https://user-images.githubusercontent.com/36833664/127934616-0aaab4c6-df60-4cb8-9ec4-8589f6274bcc.png)

**Comment by shasheene** [comment]
> why by default the Overwrite partition table is selected?

With "overwrite partition table" selected by default (along with all partitions to restore) the user can easily deselect the partitions they don't want to restore. Which is the typical behavior that almost all users will want.

Because by overwriting the partition table during a restore operation, it's guaranteed that the target partitions will match the information within the backup image. This means that the restore mapping will be correct (for example, sda4 in the backup image should be restored to /dev/sda4). Like the description says, **if you haven't resized or added new partitions since creating the backup image (eg, by installing another OS) then overwriting the partition table then overwriting the partition table won't actually have any negative impact**.

The trouble is **if you don't overwrite the partition table, then I can't easily guarantee the correct mapping. Because the destination disk might not match the source disk at all. So the user would need to configure this mapping themselves. Which definitely could be error prone**.

Basically advanced users with a large number of operating systems (and who are constantly added new operating systems) are in a position to manually map the partitions themselves (and understand the risks/benefits). But most users will not be able to configure the mapping, and won't actually need to because they aren't adding new operating systems or resizing things. So "overwrite partition table" ticked as the default makes sense for Rescuezilla.

Maybe in the future I'll find a more advance way and reliable way to generate the mapping automatically, and then no longer need to overwrite the partition table. But I haven't determined a good way yet, so simply providing the description in your screenshot is currently the best approach I have.
